### PR TITLE
Changes resulting from lazy loading controllers

### PR DESCRIPTION
**Pull request by DonJayamanne**
Part of #7583

### CODE REVIEW

File: TELEMETRY.md [hunk 1]
```diff
@@ -2180,17 +2180,7 @@ Expand each section to see more information about that event.
     Telemetry sent with the total number of different types of kernels in the kernel picker.  
     ```
 
-    - Properties:  
-        - `resourceType`?: `<see below>`
```
Comment (DonJayamanne) [inline TELEMETRY.md:2184]: This is  no longer possible as we're lazy loading the controllers

File: src/kernels/telemetry/helper.ts [hunk 2]
```diff
@@ -25,9 +25,6 @@ export type ContextualTelemetryProps = {
     kernelConnection: KernelConnectionMetadata;
     startFailed: boolean;
     kernelDied: boolean;
-    kernelSpecCount: number; // Total number of kernel specs in list of kernels.
-    kernelInterpreterCount: number; // Total number of interpreters in list of kernels
-    kernelLiveCount: number; // Total number of live kernels in list of kernels.
```
Comment (DonJayamanne) [inline src/kernels/telemetry/helper.ts:30]: No longer relevant as we're lazy loading controllers

File: src/notebooks/controllers/commands/installPythonControllerCommands.ts [hunk 3]
```diff
@@ -186,7 +186,7 @@ export class InstallPythonControllerCommands implements IExtensionSingleActivati
 
                     // Trigger a load of our notebook controllers, we want to await it here so that any in
                     // progress executions get passed to the suggested controller
-                    await this.controllerLoader.loadControllers(true);
+                    await this.controllerLoader.loaded;
```
Comment (DonJayamanne) [inline src/notebooks/controllers/commands/installPythonControllerCommands.ts:189]: We cannot refresh controllers anymore as its lazy loaded.
`loaded` property here merely means we have loaded at least one controller, which is the same as listening to the event on this class.
Created a debt issue to resolve this (doesn't feel right to have `loaded`)

File: src/notebooks/controllers/controllerLoader.ts [hunk 4]
```diff
@@ -147,11 +116,9 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
             controller.dispose(); // This should remove it from the registered list
         });
 
-        // Indicate a refresh
-        this.refreshedEmitter.fire();
-
         // Set that we have loaded controllers
-        await this.controllersLoadedContext.set(true);
+        this.controllersLoadedContext.set(true).ignoreErrors();
```
Comment (DonJayamanne) [inline src/notebooks/controllers/controllerLoader.ts:120]: @IanMatthewHuff I've created an issue for this

**Review by rebornix — APPROVED** [review]

**Pull request merged**